Re-defined strdup in toml.c

Hi!

I've included this library in a project and when trying to compile in a somewhat outdated OS it shows the following:

```
make[2]:
../src/config/toml.c:73: warning: "strdup" redefined
   73 | #define strdup(x) error:do-not-use---use-STRDUP-instead
      | 
In file included from /usr/include/string.h:630,
                 from ../src/config/toml.c:39:
/usr/include/bits/string2.h:1291: note: this is the location of the previous definition
 1291 | #   define strdup(s) __strdup (s)
```

Since it seems this is a system header, perhaps this solution would be a good workaround:
```
#undef strdup
#define strdup(x)	error:do-not-use---use-STRDUP-instead
static char* STRDUP(const char* s)
{
	int len = strlen(s);
	char* p = MALLOC(len+1);
	if (p) {
		memcpy(p, s, len);
		p[len] = 0;
	}
	return p;
}

#undef strndup
#define strndup(x)	error:do-not-use---use-STRNDUP-instead
static char* STRNDUP(const char* s, size_t n)
{
	size_t len = strnlen(s, n);
	char* p = MALLOC(len+1);
	if (p) {
		memcpy(p, s, len);
		p[len] = 0;
	}
	return p;
}
```

--- a/toml.c
+++ b/toml.c
@@ -65,6 +65,7 @@ static void* CALLOC(size_t nmemb, size_t sz)
 }
 
 
+#undef strdup
 #define strdup(x)	error:do-not-use---use-STRDUP-instead
 static char* STRDUP(const char* s)
 {
@@ -77,6 +78,7 @@ static char* STRDUP(const char* s)
 	return p;
 }
 
+#undef strndup
 #define strndup(x)	error:do-not-use---use-STRNDUP-instead
 static char* STRNDUP(const char* s, size_t n)
 {

